Lesson 4/dmitriy.pereiaslovec/nginx.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+progr_name='nginx'
+
+echo "Check for current user rights";
+if [[ $(id -u) -ne 0 ]]
+  then
+    echo "Please run as root"
+    exit 0;
+  else 
+    echo "Start executing tasks" 
+fi
+echo ""
+echo ""
+echo "------------------------"
+echo "task 1";
+echo "------------------------"
+echo ""
+echo ""
+echo "Install \"aptitude\" - program for package management";
+
+apt install --assume-yes aptitude
+
+#create list array of all installed programms in system
+readarray -t list < <( aptitude -F' * %p -> %d ' --no-gui --disable-columns search '?and(~i,!?section(libs), !?section(kernel), !?section(devel))')
+
+#alt way for print all elements from array
+#printf '%s\n' "${list[@]}"
+
+echo""
+echo "All installed programms in system:";
+echo ""
+IFS=$'\n'; echo "${list[*]}" 
+echo""
+cmd=$(printf '%s\n' "${list[@]}" | grep $progr_name );
+if [ $? = 0 ]
+then
+  echo "nginx installed in system";
+  nginx -v
+  echo "deleting from system";
+  nginx -s stop
+  apt-get --purge --assume-yes remove nginx*
+  if [ $? -eq "0" ]
+  then
+    echo "nginx fully deleted;"
+  else
+    echo "oops, something went wrong. Contact with your operations team."
+    exit 0;
+  fi  
+else
+  echo ""	
+  echo "nginx not installed in system";
+fi
+echo"";
+echo"";
+echo "------------------------"
+echo "task 2"
+echo "------------------------"
+echo"";
+echo"";
+echo "Istallation of nginx v 1.14.2"
+echo "";
+apt-get install --assume-yes curl gnupg2 ca-certificates lsb-release;
```

Comment:
Package gnupg2 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Unable to locate package curl
E: Package 'gnupg2' has no installation candidate
E: Unable to locate package ca-certificates
E: Unable to locate package lsb-release